frontend/src/layout/onboarding/onboarding.js

```diff
@@ -0,0 +1,162 @@
+                                data-attr={'onboarding-item-' + index}
+                            >
+                                <Checkbox
+                                    style={{ marginRight: 12 }}
+                                    checked={user.onboarding.steps[index] || checked[index]}
+                                ></Checkbox>
+                                <span>{value}</span>
+                            </Link>
+                        </div>
+                    )
+                })}
+                <hr style={{ height: 5, visibility: 'hidden' }} />
+                {unfinishedCount > 0 ? (
+                    <p onClick={dontShowAgain} style={{ color: 'gray', cursor: 'pointer' }}>
+                        Don't show this again
+                    </p>
+                ) : (
+                    <Button onClick={dontShowAgain} type={'primary'}>
+                        Done
+                    </Button>
+                )}
+            </div>
+        )
+    }
+
+    return (
+        <div className="mr-2 ml-2">
+            <Popover
+                visible={visible}
+                content={actionsLoading ? <Loading></Loading> : content({ actions })}
+                trigger="click"
+            >
+                <Badge count={unfinishedCount}>
+                    <Button data-attr="onboarding-button" onClick={() => (visible ? closePopup() : setVisible(true))}>
+                        {unfinishedCount === 0 ? <StarFilled></StarFilled> : <StarOutlined></StarOutlined>}
+                    </Button>
+                </Badge>
+            </Popover>
+            <Modal
+                visible={instructionalModal}
+                style={{ minWidth: '50%' }}
+                bodyStyle={{ display: 'flex', flexDirection: 'column', alignItems: 'center' }}
+                footer={null}
+                onCancel={() => setInstructionalModal(false)}
+            >
+                <img data-attr="onboarding-image" style={{ maxWidth: '100%' }} src={ModalContent[tourType].src}></img>
+                <h1 style={{ textAlign: 'center' }}>{ModalContent[tourType].title}</h1>
+                <p style={{ textAlign: 'center', fontSize: 16, fontWeight: 500, maxWidth: '60%' }}>
+                    {ModalContent[tourType].description}
+                </p>
+                <Button data-attr="onboarding-start-flow-button" type="primary" style={{ textAlign: 'center' }}>
+                    <Link
+                        to={ModalContent[tourType].link}
+                        onClick={() => {
+                            setInstructionalModal(false)
+                            closePopup()
+                            setTimeout(() => setTourActive(), 500)
+                        }}
+                    >
+                        {ModalContent[tourType].buttonText}
+                    </Link>
+                </Button>
+            </Modal>
+        </div>
+    )
+}
+
+const ModalContent = {
+    [TourType.ACTION]: {
```

Comment:
This seems to work here, but purely for thought, consider using React components instead:
```
function ModalContent(src, title, description, link, buttonText) {
   return ...extracted content here...;
}

const ModalContent = {
   [TourType.Action]: () => ModalContent(ActionImage, "Actions", "Events can get overwhelm..., ...);
   [TourType.TRENDS]: () => ModalContent(Trendimage, "Trends", .., ...);
 ....
}
```

Why? The above model is as readable, but will also adapt to change better if new content is added that does not conform to the current "structure".